packages/server/graphql/public/types/helpers/getIllustrationUrlForActivity.ts

```diff
@@ -0,0 +1,122 @@
+import getTemplateIllustrationUrl from '../../../mutations/helpers/getTemplateIllustrationUrl'
+
+const retroIllustrations: Record<string, string> = {
```

Comment:
Before figuring out the solution here, I wanted to see what the root cause was for #8713.
It looks like I wrote a migration that overwrote `illustrationUrl` for the pre/post mortem templates.
Given that knowledge, my gut reaction is to write a migration that fixes the values in the DB. That said, I know it can be frustrating to consider that data could be mutated at any time, sometimes incorrectly, and it could just be nice to have something a little more visible.

Your approach is very different! If I understand it correctly, we're nulling out all illustrationUrls in the DB & moving that data from the DB into a code file. As you explain it, this is easier to reason about because we can see the whole table at any given time. I'd like to learn more about this approach! It's nice seeing everything without having to run a query, but I'm having a hard time justifying in my head why this one column should live in code while the other columns live in the DB. For example, why not also have a lookup table for `category`? What makes the illustrationUrl special so it lives outside the DB? Now with this field living in 2 places, how might we prevent devs from forgetting to also add an entry to this table when they write a migration to add a new template?

Reply:
> why this one column should live in code while the other columns live in the DB. For example, why not also have a lookup table for category? What makes the illustrationUrl special so it lives outside the DB?

My thinking here is that this field is effectively static for any given template, so it makes sense to keep this as a static mapping that's easy to read and easy to change. Contrast this with `category` or any other template field that's mutable by the user, and couldn't easily be statically mapped.

If/when we allow users to mutate illustrations for templates they own, it might make sense to move this back into the DB, but I don't see us prioritizing that anytime soon.

> Now with this field living in 2 places, how might we prevent devs from forgetting to also add an entry to this table when they write a migration to add a new template?

This was the status quo for a while (forever?) until we moved this field to the DB. The main potential problem I see is that devs might add the illustration to the migration _instead of_ this table. I don't see a solution that would enforce this in code (unless we remove the field from the DB entirely, which would give all copied templates the custom template illustration), so I would announce the change in #engineering, and we would just need to catch this in CR if it comes up.